### PR TITLE
Putting FIFO file in new "realm.management" directory

### PR DESCRIPTION
**Pull request by austinzheng**
Per #3335.

@tgoyne @jpsim @bdash

### CODE REVIEW

**Comment by tgoyne** [comment]
A file named just ".note" is weird. Should be something like "notification_pipe".

Moving this file breaks sharing the realm file with previous versions of Realm. We haven't really decided if we consider that a breaking change.

**Comment by austinzheng** [comment]
I think I took the ticket description too literally. I'll fix the file name.

**Comment by austinzheng** [comment]
Is there anything I can do to move this ticket forward? If I create a symlink in the original location will that be enough to get it approved?

File: Realm/ObjectStore/impl/apple/external_commit_helper.cpp [hunk 1]
```diff
@@ -95,10 +95,20 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
     }
 
 #if !TARGET_OS_TV
-    auto path = parent.get_path() + ".note";
+    auto dir_path = parent.get_path() + ".management/";
+    auto path = dir_path + "notification_pipe.note";
```
Comment (jpsim) [inline Realm/ObjectStore/impl/apple/external_commit_helper.cpp]: does this need an extension? Weird to call it "note{...}.note". Why not just `notification_pipe`?

**Comment by jpsim** [comment]
A symlink in the current location seems sensible as a transition step. Seems like we ought to find a way to make this relevant to users in the changelog too.

**Comment by jpsim** [comment]
Putting this "on hold" as it hasn't been actively maintained.

**Comment by austinzheng** [comment]
Closing this PR since it would be far less work to just redo it from scratch than to try rebasing it onto HEAD.

**Pull request closed without merging**